model/src/test/kotlin/licenses/ResolvedLicenseInfoTest.kt

```diff
@@ -143,6 +149,177 @@ class ResolvedLicenseInfoTest : WordSpec() {
+                    Identifier("Maven", "org.oss-review-toolkit", "example", "0.0.1"),
+                    violations,
+                    resolutions
+                )
+
+                filteredLicenses should haveSize(1)
+                filteredLicenses[0].license shouldBe SpdxSingleLicenseExpression.parse(bsd)
+            }
+
+            "not remove license that does not contain a license rule violation resolution" {
+                val resolvedLicenses = listOf(
+                    ResolvedLicense(
+                        license = gpl.toSpdx() as SpdxSingleLicenseExpression,
+                        originalDeclaredLicenses = setOf(gpl),
+                        originalExpressions = mapOf(
+                            LicenseSource.DECLARED to setOf(gpl.toSpdx()),
+                            LicenseSource.DETECTED to setOf(gpl.toSpdx())
+                        ),
+                        locations = emptySet()
+                    ),
+                    ResolvedLicense(
+                        license = bsd.toSpdx() as SpdxSingleLicenseExpression,
+                        originalDeclaredLicenses = setOf(bsd),
+                        originalExpressions = mapOf(
+                            LicenseSource.DECLARED to setOf(bsd.toSpdx()),
+                            LicenseSource.DETECTED to setOf(bsd.toSpdx())
+                        ),
+                        locations = emptySet()
+                    ),
+                )
+
+                val violations = listOf(
+                    RuleViolation(
+                        "example rule",
+                        Identifier("Maven", "org.oss-review-toolkit", "example", "0.0.1"),
+                        SpdxSingleLicenseExpression.parse(gpl),
+                        null,
+                        Severity.ERROR,
+                        "example message",
+                        "how to fix"
+                    ),
+                    RuleViolation(
+                        "example rule",
+                        Identifier("Maven", "org.oss-review-toolkit", "example", "0.0.10"),
+                        SpdxSingleLicenseExpression.parse(bsd),
+                        null,
+                        Severity.ERROR,
+                        "example message",
+                        "how to fix"
+                    ),
+                )
+
+                val resolutions = listOf(
+                    RuleViolationResolution(
+                        "different regex", RuleViolationResolutionReason.CANT_FIX_EXCEPTION, "example comment"
+                    )
+                )
+
+                val filteredLicenses = resolvedLicenses.filterLicenseResolutions(
+                    Identifier("Maven", "org.oss-review-toolkit", "example", "0.0.1"),
+                    violations,
+                    resolutions
+                )
+
+                filteredLicenses should haveSize(2)
+            }
+        }
```

Comment:
You could test additionally the corner case that no licenses are filtered if the list of violations is empty.